src/include/OpenImageIO/span.h

```diff
@@ -143,11 +143,7 @@ class span {
         : span (il.begin(), il.size()) { }
 
     /// Assignment copies the pointer and length, not the data.
-    span& operator= (const span &copy) {
-        m_data = copy.data();
-        m_size = copy.size();
-        return *this;
-    }
+    constexpr span& operator= (const span &copy) = default;
```

Comment:
This change is pretty close to what [std::span does](https://en.cppreference.com/w/cpp/container/span/operator%3D). The only bit missing, if you want it to fully match, is marking it noexcept:
```c++
constexpr span& operator=( const span& other ) noexcept = default;
```

Reply:
You're right. And I noticed a few other spots where we can be noexcept. Testing now and will amend the PR if it works.

Reply:
updated